Docker/Runner/Limits.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace Keboola\DockerBundle\Docker\Runner;
+
+use Keboola\DockerBundle\Docker\Image;
+use Keboola\Syrup\Exception\ApplicationException;
+use Psr\Log\LoggerInterface;
+
+class Limits
+{
+    /**
+     * @var array
+     */
+    private $userFeatures;
+
+    /**
+     * @var array
+     */
+    private $projectFeatures;
+
+    /**
+     * @var array
+     */
+    private $projectLimits;
+
+    /**
+     * @var array
+     */
+    private $instanceLimits;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * Limits constructor.
+     * @param LoggerInterface $logger
+     * @param array $instanceLimits
+     * @param array $projectLimits
+     * @param array $projectFeatures
+     * @param array $userFeatures
+     */
+    public function __construct(
+        LoggerInterface $logger,
+        array $instanceLimits,
+        array $projectLimits,
+        array $projectFeatures,
+        array $userFeatures
+    ) {
+        $this->logger = $logger;
+        $this->instanceLimits = $instanceLimits;
+        $this->projectLimits = $projectLimits;
+        $this->projectFeatures = $projectFeatures;
+        $this->userFeatures = $userFeatures;
+    }
+
+    public function getMemoryLimit(Image $image)
+    {
+        return $image->getSourceComponent()->getMemory();
+    }
+
+    public function getMemorySwapLimit(Image $image)
+    {
+        return $image->getSourceComponent()->getMemory();
+    }
+
+    public function getCpuSharesLimit(Image $image)
+    {
+        return $image->getSourceComponent()->getCpuShares();
+    }
+
+    public function getNetworkLimit(Image $image)
+    {
+        return $image->getSourceComponent()->getNetworkType();
+    }
+
+    public function getCpuLimit(Image $image)
+    {
+        return $this->getInstanceCpuLimit();
+    }
+
+    private function getInstanceCpuLimit()
+    {
+        if (isset($this->instanceLimits['cpu_count']) &&
+            filter_var($this->instanceLimits['cpu_count'], FILTER_VALIDATE_INT,
```

Comment:
co na to použít Symfony Validator? http://api.symfony.com/3.4/Symfony/Component/Validator/Constraints.html

Reply:
abych pravdu rekl vubec nevim jak to pouzit

Reply:
cca takhle
```
$validator = \Symfony\Component\Validator\Validation::createValidator();

$violations = $validator->validate(1, array(
    new \Symfony\Component\Validator\Constraints\Range(array('min' => 10, 'max' => 20)),
));

var_dump($violations);
```

Reply:
aha, jo tak nakonec je to docela mily, ale z dokumentace jsem to teda moc nepobral, asi nejaky skvrny na mozku

Reply:
ne, žádný skrvyn, jsem si to musel rozběhat lokálně a hrát si s tím.